core/system/inc/svc_gw.h

```diff
@@ -39,8 +39,7 @@ static inline void svc_gw_check_magic(TSecGw *svc_pc)
         HALT_ERROR(SANITY_CHECK_FAILED,
                    "secure gateway not in flash (0x%08X)", svc_pc);
 
-    if((svc_pc->opcode & ~((uint16_t) UVISOR_SVC_CUSTOM_MSK)) !=
-        UVISOR_SVC_GW_OPCODE)
+    if((svc_pc->opcode & ~((uint16_t) UVISOR_SVC_FAST_NARGS_MASK)) != UVISOR_SVC_GW_OPCODE)
```

Comment:
Because the full opcode is 16 bits, while the `NARGS` mask is 8 bits (only covers imm8 field of `svc`).

If `NARGS_MASK` is `0x70` (8 bits) I want it to become `0xFF8F` (16 bits).